src/main/java/org/llorllale/cactoos/matchers/Throws.java

```diff
@@ -66,35 +82,76 @@
      * @param msg The expected exception message.
      * @param type The expected exception type.
      */
-    public Throws(final String msg, final Class<? extends Exception> type) {
+    public Throws(
+        final String msg, final Class<? extends Exception> type
+    ) {
+        this(
+            type,
+            actual -> actual.equals(msg),
+            new UncheckedText(
+                new FormattedText("message '%s'", msg)
+            ).asString()
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param type The expected exception type.
+     * @param expectation The expectation which the exception message must
+     *  match.
+     */
+    public Throws(
+        final Class<? extends Exception> type,
+        final Func<String, Boolean> expectation
+    ) {
+        this(type, expectation, "expectation for message.");
+    }
+
+    /**
+     * Ctor.
+     * @param type The expected exception type.
+     * @param messageExpectation The expectation which the exception message
+     *  must match.
+     * @param expectationDescription The description of the message
+     *  expectation.
+     */
+    public Throws(
+        final Class<? extends Exception> type,
+        final Func<String, Boolean> messageExpectation,
```

Comment:
@vzurauskas parameter `messageExpectation` hides `this.messageExpectation`

Reply:
@ilyakharlamov Same reply.